### PR TITLE
Don't call length if not needed

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/pars.fsy [hunk 1]
```diff
@@ -2403,8 +2409,11 @@ hardwhiteLetBindings:
         BindingSetPreAttrs(mLetKwd,isRec,isUse,
           (fun attrs vis -> 
             let binds = localBindingsBuilder attrs vis mLetKwd 
-            if not isRec && List.length binds > 1 then 
-                    reportParseErrorAt mLetKwd (FSComp.SR.parsLetAndForNonRecBindings()) 
+            if isRec then [],binds else
```
Comment (dsyme) [inline src/fsharp/pars.fsy]: I'd prefer a new list function ``List.isNilOrSingleton`` so

    if not isRec && not (List.isNilOrSingleton binds) then ...

Comment (forki) [inline src/fsharp/pars.fsy]: good idea

Comment (dsyme) [inline src/fsharp/pars.fsy]: Or ``List.hasLengthGreaterThen 1``

File: src/fsharp/MethodOverrides.fs [hunk 2]
```diff
@@ -305,12 +305,12 @@ module DispatchSlotChecking =
...
                         | [ Override(_,_,_,(mtps,_),argTys,_,_,_) as overrideBy ] ->
-                            let possibleDispatchSlots =
+                            let notMoreThanOnePossibleDispatchSlot =
                                 dispatchSlots
                                 |> List.filter (fun (RequiredSlot(dispatchSlot,_)) -> IsNameMatch dispatchSlot overrideBy && IsImplMatch g dispatchSlot overrideBy)
-                                |> List.length
+                                |> isNilOrSingleton
                            
-                            if possibleDispatchSlots > 1 then
+                            if not notMoreThanOnePossibleDispatchSlot then
```
Comment (KevinRansom) [inline src/fsharp/MethodOverrides.fs]: No!!!! ...  
It took a third look for me to release that this said __not not....__ so a better name for __notMoreThanOnePossibleDispatchSlot__ please.

Comment (forki) [inline src/fsharp/MethodOverrides.fs]: good point. fixed

**Pull request merged**